src/sentry/tasks/groupowner.py

```diff
@@ -0,0 +1,111 @@
+        else:
+            project = Project.objects.get_from_cache(id=event.project_id)
+            owners = GroupOwner.objects.filter(
+                group_id=event.group_id,
+                project=project,
+                organization_id=project.organization_id,
+                type=GroupOwnerType.SUSPECT_COMMIT.value,
+            )
+            owner_count = owners.count()
+            if owner_count >= PREFERRED_GROUP_OWNERS:
+                # We have enough owners already - so see if any are old.
+                # If so, we can delete it and replace with a fresh one.
+                owners = owners.filter(
+                    date_added__lte=timezone.now() - PREFERRED_GROUP_OWNER_AGE
+                ).order_by("-date_added")
+                if not owners.exists():
+                    metrics.incr(
+                        "sentry.tasks.process_suspect_commits.aborted",
+                        tags={"detail": "maxed_owners_none_old"},
+                    )
+                    can_process = False
+
+            owner_data = {"count": owner_count, "time": timezone.now()}
+            cache.set(cache_key, owner_data, OWNER_CACHE_LIFE)
+
+        if can_process:
+            with metrics.timer("sentry.tasks.process_suspect_commits.process_loop"):
+                metrics.incr("sentry.tasks.process_suspect_commits.calculated")
+                try:
+
+                    with metrics.timer(
+                        "sentry.tasks.process_suspect_commits.get_serialized_event_file_committers"
+                    ):
+                        committers = get_event_file_committers(project, event)
+                    # TODO(Chris F.) We would like to store this commit information so that we can get perf gains
+                    # and synced information on the Issue details page.
+                    # There are issues with this...like mutable commits and commits coming in after events.
+                    for committer in committers:
+                        if "id" in committer["author"]:
+                            owner_id = committer["author"]["id"]
+                            go, created = GroupOwner.objects.update_or_create(
+                                group_id=event.group_id,
+                                type=GroupOwnerType.SUSPECT_COMMIT.value,
+                                user_id=owner_id,
+                                project=project,
+                                organization_id=project.organization_id,
+                                defaults={
+                                    "date_added": timezone.now()
+                                },  # Updates date of an existing owner, since we just matched them with this new event,
+                            )
+                            if created:
+                                owner_count += 1
+                                if owner_count > PREFERRED_GROUP_OWNERS:
+                                    owners.first().delete()
```

Comment:
If we have 4 committers, then isn't this going to delete committers multiple times unnecessarily? So we have commiters A, B, C, D.

First run:
 - We check A, create the row, no deletions
 - Check B, create the row, no deletions
 - Check C, create the row, delete A
 - Check D, create the row, delete B

We should probably cap the number of committers that we process to `PREFERRED_GROUP_OWNERS` to avoid this?

Reply:
Oops, meant to short circuit this loop last time.